apps/web/components/dialog/RerouteDialog.tsx

```diff
@@ -469,6 +471,7 @@ const NewRoutingManager = ({
     // TODO: Long term, we should refactor handleNewBooking and use a different route specific for this purpose,
     createBookingMutation.mutate({
       rescheduleUid: booking.uid,
+      rescheduledBy: session?.data?.user?.email ?? undefined,
```

Comment:
Shouldn't we error early if session is nullish, because rerouting can't be done by a non-logged in user. 

Doing this change make it feel like non-logged in user could do re-routing

Reply:
A non-logged in user would never see this dialog because we only show this for authed users who have access to make the reroute. The typing just makes this undefined.

Reply:
Yeah, but if he is never supposed to see this dialog, then I think we should make it explict and check for session the first thing in the component(after all the hooks) and early return.

Then we could just do session.data.user.email here. It avoids the entire component to keep executing in such a case.